src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1810,9 +1810,29 @@ internal Symbol ContainingMember()
             return containingMember;
         }
 
-        private BoundExpression TryBindInteractiveReceiver(SyntaxNode syntax, Symbol currentMember, NamedTypeSymbol currentType, NamedTypeSymbol memberDeclaringType)
+        private bool IsInstanceContext()
         {
-            if (currentType.TypeKind == TypeKind.Submission && currentMember.RequiresInstanceReceiver())
+            var containingMember = this.ContainingMemberOrLambda;
+            do
+            {
+                if (containingMember.IsStatic)
+                {
+                    return false;
+                }
+                if (containingMember.Kind == SymbolKind.NamedType)
+                {
+                    break;
+                }
+                containingMember = containingMember.ContainingSymbol;
+            } while ((object)containingMember != null);
+            return true;
+        }
+
+        private BoundExpression TryBindInteractiveReceiver(SyntaxNode syntax, NamedTypeSymbol currentType, NamedTypeSymbol memberDeclaringType)
```

Comment:
>NamedTypeSymbol currentType [](start = 78, length = 27)

I think this parameter should be removed. Why force consumers to understand what to pass here and make it possible to make a mistake?